misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/Formatter.java

```diff
@@ -0,0 +1,113 @@
+import io.ballerinalang.compiler.text.LineRange;
+import io.ballerinalang.compiler.text.TextDocument;
+import io.ballerinalang.compiler.text.TextDocuments;
+
+/**
+ * Class that exposes the formatting APIs.
+ */
+public class Formatter {
+
+    /**
+     * Formats the provided source string and returns back the formatted source string.
+     *
+     * @param source A Ballerina source in string form
+     * @return A modified source string after formatting changes
+     */
+    public static String format(String source) {
+        return format(source, new FormattingOptions());
+    }
+
+    /**
+     * Formats a line range of the provided SyntaxTree. All the nodes falling within the line range
+     * specified will be formatted.
+     *
+     * @param syntaxTree The complete SyntaxTree, of which a part is to be formatted
+     * @param range LineRange which specifies the range to be formatted
+     * @return The modified SyntaxTree after formatting changes
+     */
+    public static SyntaxTree format(SyntaxTree syntaxTree, LineRange range) {
+        return format(syntaxTree, range, new FormattingOptions());
+    }
+
+    /**
+     * Formats the provided SyntaxTree and returns back a formatted SyntaxTree.
+     *
+     * @param syntaxTree The SyntaxTree which is to be formatted
+     * @return The modified SyntaxTree after formatting changes
+     */
+    public static SyntaxTree format(SyntaxTree syntaxTree) {
+        return format(syntaxTree, new FormattingOptions());
+    }
+
+    /**
+     * Formats the provided source string while using the formatting options provided.
+     *
+     * @param source A Ballerina source in string form
+     * @param options Formatting options that are to be used when formatting
+     * @return A modified source string after formatting changes
+     */
+    public static String format(String source, FormattingOptions options) {
+        TextDocument textDocument = TextDocuments.from(source);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        FormattingTreeModifier treeModifier = new FormattingTreeModifier();
+        ModulePartNode newModulePart = treeModifier.transform((ModulePartNode) syntaxTree.rootNode());
+        if (options != null) {
+            treeModifier.setFormattingOptions(options);
+        }
+        return syntaxTree.modifyWith(newModulePart).toSourceCode();
+    }
+
+    /**
+     * Formats a line range of the provided SyntaxTree while using the formatting options provided. All the
+     * nodes falling within the line range provided will be formatted.
+     *
+     * @param syntaxTree The complete SyntaxTree, of which a part is to be formatted
+     * @param range LineRange which needs to be formatted
+     * @param options Formatting options that are to be used when formatting
+     * @return The modified SyntaxTree after formatting changes
+     */
+    public static SyntaxTree format(SyntaxTree syntaxTree, LineRange range, FormattingOptions options) {
```

Comment:
```suggestion
    private static SyntaxTree format(SyntaxTree syntaxTree, LineRange range, FormattingOptions options) {
```

Reply:
This is an API exposed by the formatter core, it is to be used by other modules such as the lang-sever and the formatter CLI.